pkg/meta/base.go

```diff
@@ -2348,6 +2339,35 @@ func (m *baseMeta) checkTrash(parent Ino, trash *Ino) syscall.Errno {
 	return st
 }
 
+func (m *baseMeta) preCreateTrash() {
+	utils.SleepWithJitter(10 * time.Second) // Avoid txn conflicts caused by concurrent creation
+	name := time.Now().Add(time.Hour).UTC().Format("2006-01-02-15")
+	var nextTrash Ino
+	if st := m.ensureTrash(name, &nextTrash); st != 0 {
+		logger.Warnf("Create next trash %s: %s", name, st)
+	} else {
+		logger.Debugf("Created next trash %s, inode %d", name, nextTrash)
+	}
+}
+
+func (m *baseMeta) ensureTrash(name string, trash *Ino) syscall.Errno {
+	st := m.en.doLookup(Background, TrashInode, name, trash, nil)
+	if st == syscall.ENOENT {
+		next, err := m.en.incrCounter("nextTrash", 1)
+		if err == nil {
+			*trash = TrashInode + Ino(next)
+			attr := Attr{Typ: TypeDirectory, Nlink: 2, Length: 4 << 10, Parent: TrashInode, Full: true}
+			st = m.en.doMknod(Background, TrashInode, name, TypeDirectory, 0555, 0, "", trash, &attr)
```

Comment:
Maybe we can move nextTrash into doMknod() to avoid the conflict?

Reply:
Do you mean moving `next, err := m.en.incrCounter("nextTrash", 1)` into `m.en.doMknod`? then conflict will happen in `doMknod`, as it becomes a larger txn.

After applying the prefetch and `utils.SleepWithJitter` in our environment. I never see a conflict of `nextTrash` again.

Reply:
Once a client succeed, it will not try to increase nextTrash, so you will not see retried transaction.